### PR TITLE
feat: add benchmarking framework

### PR DESCRIPTION
**Pull request by phc1990**
Adding benchmarking framework
- Added https://github.com/benchmark-action/github-action-benchmark and tailored for a C++ project using Google's benchmark
- Added propagation and access generation as initial benchmark scenarios
- Added pipeline alerts whenever benchmarks deteriorate by more than 5%
- Added metrics publishing (when branch is `main`) to [GitHub pages](https://open-space-collective.github.io/open-space-toolkit-astrodynamics/dev/bench/) (current values from previous test commits).

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -49,3 +49,39 @@ jobs:
...
+        with:
+          fetch-depth: 0
+          lfs: true
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Pull Development Image
+        run: docker pull openspacecollective/${{ needs.prepare-environment.outputs.project_name }}-development:${{ needs.prepare-environment.outputs.project_version }}
```
Comment (phc1990) [inline .github/workflows/build-test.yml:71]: I had to hardcode the REGISTRY_OATH (`openspacecollective`) as I was not able to get it from `build-and-test` `env` section.

Any ideas?

Comment (vishwa2710) [reply .github/workflows/build-test.yml:71]: I think that's reasonable?

File: CMakeLists.txt [hunk 2]
```diff
@@ -445,6 +446,44 @@ IF (BUILD_UNIT_TESTS)
...
+
+    ### benchmark
+
+    FIND_PACKAGE ("benchmark" REQUIRED)
+
+    TARGET_INCLUDE_DIRECTORIES (${BENCHMARK_TARGET} PUBLIC ${benchmark_INCLUDE_DIRS})
+    TARGET_INCLUDE_DIRECTORIES (${BENCHMARK_TARGET} PUBLIC "${PROJECT_SOURCE_DIR}/include")
+    TARGET_INCLUDE_DIRECTORIES (${BENCHMARK_TARGET} PUBLIC "${PROJECT_SOURCE_DIR}/benchmark")
+
+    # TBM: Don't hardcode this
```
Comment (vishwa2710) [inline CMakeLists.txt:475]:
```suggestion
```

**Review by vishwa2710 — APPROVED** [review]
Yo this looks sick! I think we just have to update the code coverage command to ignore the benchmark.cpp files and we're golden. In a future MR it would be great to add python benchmarks as well!

**Pull request merged**